app/src/main/java/com/maq/xprize/onecourse/hindi/utils/Zip.java

```diff
@@ -100,19 +101,23 @@ public void run() {
                     while ((currByte = inputStream.read()) != -1) {
                         outputStream.write(currByte);
                     }
-                    if (isMain) {
-                        editor.putInt("mainFileVersion", expansionFileVersion);
-                    } else {
-                        editor.putInt("patchFileVersion", expansionFileVersion);
-                    }
-                    editor.commit();
+                    isExpansionSuccessful = true;
                 } catch (Exception e) {
                     e.printStackTrace();
+                    isExpansionSuccessful = false;
                 } finally {
                     outputStream.close();
                     inputStream.close();
                 }
             }
         }
+        if (isExpansionSuccessful) {
+            if (isMain) {
+                editor.putInt("mainFileVersion", expansionFileVersion);
+            } else {
+                editor.putInt("patchFileVersion", expansionFileVersion);
+            }
```

Comment:
__editor.putInt(isMain? "mainFileVersion" : "patchFileVersion", expansionFileVersion)__
User ternary operator for readability

Reply:
Done